libs/sdk-core/src/breez_services.rs

```diff
@@ -1220,55 +1220,90 @@ impl BreezServices {
+            .and_then(|s| s.block_time)
+        {
+            None => {
+                warn!("Blocktime could not be determined for from closing outspend, defaulting closed_at to epoch time");
+                SystemTime::now().duration_since(UNIX_EPOCH)?.as_secs()
+            }
+            Some(block_time) => block_time,
+        };
+        Ok(closed_at)
+    }
+
+    async fn lookup_chain_service_closing_txid(
+        &self,
+        channel: crate::models::Channel,
+    ) -> Result<Option<String>> {
+        let maybe_closing_txid = self
+            .lookup_chain_service_closing_outspend(channel.clone())
+            .await?
+            .and_then(|outspend| outspend.txid);
+        Ok(maybe_closing_txid)
+    }
+
     async fn closed_channel_to_transaction(
         &self,
         channel: crate::models::Channel,
     ) -> Result<Payment> {
-        let now_epoch_sec = SystemTime::now().duration_since(UNIX_EPOCH)?.as_secs();
+        let mut persist_channel_updates = false;
 
-        let channel_closed_at = match channel.closed_at {
+        let closed_at = match channel.closed_at {
             Some(closed_at) => closed_at,
             None => {
-                // If we don't have it, we look it up from the channel closing tx
-                let looked_up_channel_closed_at = match channel.funding_outnum {
-                    None => {
-                        warn!("No founding_outnum found for the closing tx, defaulting closed_at to epoch time");
-                        now_epoch_sec
-                    }
-                    Some(outnum) => {
-                        // Find the output tx that was used to fund the channel
-                        let outspends = self
-                            .chain_service
-                            .transaction_outspends(channel.funding_txid.clone())
-                            .await?;
-                        let maybe_block_time = outspends
-                            .get(outnum as usize)
-                            .and_then(|outspend| outspend.status.as_ref())
-                            .and_then(|status| status.block_time);
-
-                        match maybe_block_time {
-                            None => {
-                                warn!("Blocktime could not be determined for funding_outnum {outnum}, defaulting closed_at to epoch time");
-                                now_epoch_sec
-                            }
-                            Some(block_time) => block_time,
-                        }
-                    }
-                };
-
-                // Persist closed_at, if we had to look it up
-                let mut updated_channel = channel.clone();
-                updated_channel.closed_at = Some(looked_up_channel_closed_at);
-                self.persister.insert_or_update_channel(updated_channel)?;
+                persist_channel_updates = true;
+                self.lookup_chain_service_closed_at(channel.clone()).await?
```

Comment:
At this point we need to fetch also the closing tx id so it looks better to make this function `lookup_chain_service_closed_at` return both the closed_at and the txid instead of calling it twice and save the network call.

Reply:
Done, I merged the two lookups in one method.